plugin/src/main/java/org/opensearch/ml/sdkclient/LocalClusterIndicesClient.java

```diff
@@ -80,13 +81,14 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRe
         return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<PutDataObjectResponse>) () -> {
             try (XContentBuilder sourceBuilder = XContentFactory.jsonBuilder()) {
                 log.info("Indexing data object in {}", request.index());
-                IndexResponse indexResponse = client
-                    .index(
-                        new IndexRequest(request.index())
-                            .setRefreshPolicy(IMMEDIATE)
-                            .source(request.dataObject().toXContent(sourceBuilder, EMPTY_PARAMS))
-                    )
-                    .actionGet();
+                IndexRequest indexRequest = new IndexRequest(request.index())
+                    .opType(request.overwriteIfExists() ? OpType.INDEX : OpType.CREATE)
```

Comment:
Should we set default value to `True`? In OS, default value of the index request is `OpType.INDEX;`

Reply:
The default is set to true in the PutObjectRequestBuilder.
```java
    /**
     * Class for constructing a Builder for this Request Object
     */
    public static class Builder {
        private String index = null;
        private String id = null;
        private String tenantId = null;
        private boolean overwriteIfExists = true;
        private ToXContentObject dataObject = null;
```

Reply:
Oh yeah, I missed that.